form-repository.ts

```diff
@@ -28,21 +23,113 @@ export const fetchFormDefinitionById = async function (
     SELECT ?formDefinition ?formTtl
     WHERE {
       ?formDefinition a ext:GeneratedForm ;
-        mu:uuid ${sparqlEscapeString(id)} ;
+        mu:uuid ${sparqlEscapeString(formId)} ;
         ext:ttlCode ?formTtl .
     } LIMIT 1
   `);
 
   if (result.results.bindings.length) {
     const binding = result.results.bindings[0];
-    return {
-      formTtl: binding.formTtl.value,
-    };
+    return binding.formTtl.value;
   } else {
     return null;
   }
 };
 
+// Caution: form:options ?object contains a JSON, not a URI
+const buildSelectConceptSchemeUriQuery = () =>
+  `
+  PREFIX form: <http://lblod.data.gift/vocabularies/forms/>
+
+  SELECT ?o
+  WHERE {
+    ?s form:options ?o
+  }
+  `;
+
+const formOptionsToConceptSchemeUri = (binding): string => {
+  const formOptionsJson: string = binding.get('o').value;
+  const { conceptScheme } = JSON.parse(formOptionsJson);
+  return conceptScheme;
+};
+
+const fetchConceptSchemeUris = async (formTtl: string): Promise<string[]> => {
+  const query = buildSelectConceptSchemeUriQuery();
+  const store = await ttlToStore(formTtl);
+  const bindings = await queryStore(query, store);
+
+  return bindings.map(formOptionsToConceptSchemeUri);
+};
+
+const buildConstructConceptSchemesQuery = (
+  conceptSchemeUris: string[],
+): string => {
+  const buildPattern = (conceptSchemeUri: string): string =>
+    `
+    { ?s skos:topConceptOf ${sparqlEscapeUri(conceptSchemeUri)}.
+    ?s ?p ?o }
+    `;
+
+  const patterns = conceptSchemeUris.map(buildPattern).join('\nUNION\n');
+
+  return `
+    PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+  
+    CONSTRUCT {
+      ?s ?p ?o
+    } WHERE {
+      ${patterns}
+    }
+    `;
+};
+
+const sparqlEscapeObject = (bindingObject): string =>
```

Comment:
since your function is multi-line, i would really use braces and be more explicit. Two more keystrokes, but you get a lot more readability in return :raised_hands:

Reply:
Prettier disagrees, braces immediately get removed